File: workspace/1_InterleaveInString/test/util/InterleaveInStringTest.java
package util;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;

public class InterleaveInStringTest {

	@Test
	public void interleaveInEmptyString() {
		assertEquals("", StringUtil.interleave("", 'a', 2));
	}
	
	@Test
	public void interleavingAtPositionLargerThanInterleaveInterval() {
		assertEquals("hallo", StringUtil.interleave("hallo", 'a', 7));
	}
	
	@Test
	public void interleavingExactlyAtInterleaveInterval() {
		assertEquals("halloa", StringUtil.interleave("hallo", 'a', 5));
	}
	
	@Test
	public void interleavingMultipleTimes() {
		assertEquals("blablabla", StringUtil.interleave("blblbl", 'a', 2));
	}
}

File: solution/1_InterleaveInString/1_InterleaveInString/InterleaveInStringTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using util;

namespace test
{
    [TestClass]
    public class InterleaveInStringTest
    {
        [TestMethod]
        public void interleaveInEmptyString() {
		    Assert.AreEqual("", StringUtil.interleave("", 'a', 2));
	    }
	
	    [TestMethod]
	    public void interleavingAtPositionLargerThanInterleaveInterval() {
            Assert.AreEqual("hallo", StringUtil.interleave("hallo", 'a', 7));
	    }
	
	    [TestMethod]
	    public void interleavingExactlyAtInterleaveInterval() {
            Assert.AreEqual("halloa", StringUtil.interleave("hallo", 'a', 5));
	    }

        [TestMethod]
	    public void interleavingMultipleTimes() {
            Assert.AreEqual("blablabla", StringUtil.interleave("blblbl", 'a', 2));
	    }
    }
}
